File: type-safe-sync/src/main/scala/templemore/sync/scalatypegoodness/ContentRepository.scala
package templemore.sync.scalatypegoodness

/**
 * @author Chris Turner
 */
class ContentRepository {

  def findStoryById(id: String) = Some(new Story(id));
  def findImageById(id: String) = Some(new Image(id));
  def findBinaryObjectById(id: String) = Some(new BinaryObject(id));
}

File: type-safe-sync/src/main/java/templemore/sync/javaoriginal/ContentRepository.java
package templemore.sync.javaoriginal;

/**
 * @author Chris Turner
 */
public class ContentRepository {

    public Story findStoryById(final String id) {
        return new Story(id);
    }

    public Image findImageById(final String id) {
        return new Image(id);
    }

    public BinaryObject findBinaryObjectById(final String id) {
        return new BinaryObject(id);
    }
}
